File: DKR1/res.py
import math
def less_than_minSeven(x):
    y = x**(1/3)/(math.cos(x) + math.sin(x))
    return y
    
def mineSeven_less_x_less_zero(x):
    y = (math.log(abs(x),10)) + x 
    return round(y,2) 
    
def x_more_than_zero(x):
    y = math.tan(x)/(math.cos(2*x)) - (math.cos(x) / math.tan(x))
    return round(y,2)

x = float(input('Введите число x'))

if x < -7:
    y = less_than_minSeven(x)
    print(y)
if x >= -7 and x < 0:
    print('Вы ввели отрицательное число,но формула автоматически посчитала модуль этого числа')
    y = mineSeven_less_x_less_zero(x)

    print(y)
if x >= 0:
    y = x_more_than_zero(x)
    print(y)
    
start = -9
stop = 2 
step = 0.3

x_val = []
reslt = []
x = start
while x<stop:
    x_val.append(round(x,2))
    if x<-7:
        reslt.append(less_than_minSeven(x))
    elif -7<= x<0:
        reslt.append(mineSeven_less_x_less_zero(x))
    else:
        reslt.append(x_more_than_zero(x))
    x+=step
print('\nЗначение функции на интервале от -9 до 2')
for x_v,res in zip(x_val,reslt):
    print(f'f({x_v}) = {res}')
    
    



    
    

File: DKR1/res.pas
program my52;


function LessThanMinSeven(x: Real): Real;
begin
    LessThanMinSeven := Power(x, 1/3) / (Cos(x) + Sin(x));
end;

function MineSevenLessXLessZero(x: Real): Real;
begin
    MineSevenLessXLessZero := Log10(Abs(x)) + x;
end;

function XMoreThanZero(x: Real): Real;
begin
    XMoreThanZero := Tan(x) / Cos(2 * x) - (Cos(x) / Tan(x));
end;

var
    x, y: Real;
    start, stop, step: Real;
    x_val: array of Real;
    reslt: array of Real;
    i: Integer;

begin
    Write('Введите число x: ');
    ReadLn(x);

    if x < -7 then
    begin
        y := LessThanMinSeven(x);
        WriteLn(y:0:2);
    end
    else if (x >= -7) and (x < 0) then
    begin
        WriteLn('Вы ввели отрицательное число, но формула автоматически посчитала модуль этого числа');
        y := MineSevenLessXLessZero(x);
        WriteLn(y:0:2);
    end
    else
    begin
        y := XMoreThanZero(x);
        WriteLn(y:0:2);
    end;

    start := -9;
    stop := 2;
    step := 0.3;
    SetLength(x_val, Trunc((stop - start) / step) + 1);
    SetLength(reslt, Trunc((stop - start) / step) + 1);

    i := 0;
    x := start;
    while x < stop do
    begin
        x_val[i] := round(x);
        if x < -7 then
            reslt[i] := LessThanMinSeven(x)
        else if (x >= -7) and (x < 0) then
            reslt[i] := MineSevenLessXLessZero(x)
        else
            reslt[i] := XMoreThanZero(x);
        
        x := x + step;
        i := i + 1;
    end;

    WriteLn('Значение функции на интервале от -9 до 2');
    for i := 0 to Length(x_val) - 1 do
    begin
        WriteLn('f(', x_val[i]:0:2, ') = ', reslt[i]:0:2);
    end;
end.
